models/glif_psc.h

```diff
@@ -0,0 +1,354 @@
+  friend class nest::UniversalDataLogger< glif_psc >;
+
+
+  struct Parameters_
+  {
+    double G_;               // membrane conductance in nS
+    double E_L_;             // resting potential in mV
+    double th_inf_;          // infinity threshold in mV
+    double C_m_;             // capacitance in pF
+    double t_ref_;           // refractory time in ms
+    double V_reset_;         // Membrane voltage following spike in mV
+    double a_spike_;         // threshold additive constant following reset in mV
+    double b_spike_;         // spike induced threshold in 1/ms
+    double voltage_reset_a_; // voltage fraction following reset coefficient
+    double voltage_reset_b_; // voltage additive constant following reset in mV
+    double a_voltage_;       // a 'leak-conductance' for the voltage-dependent
+                             // component of the threshold in 1/ms
+    double b_voltage_;       // inverse of which is the time constant of the
+    // voltage-dependent component of the threshold in 1/ms
+    std::vector< double > asc_init_; // initial values of ASCurrents_ in pA
+    std::vector< double > k_;        // predefined time scale in 1/ms
+    std::vector< double > asc_amps_; // in pA
+    std::vector< double > r_;        // coefficient
+    std::vector< double > tau_syn_;  // synaptic port time constants in ms
+
+    // boolean flag which indicates whether the neuron has connections
+    bool has_connections_;
+
+    model_type glif_model_;
+
+    size_t n_receptors_() const; //!< Returns the size of tau_syn_
+
+    Parameters_();
+
+    void get( DictionaryDatum& ) const;
+    double set( const DictionaryDatum& );
+  };
+
+
+  struct State_
+  {
+    double U_;                         // relative membrane potential in mV
+    std::vector< double > ASCurrents_; // after-spike currents in pA
+    double ASCurrents_sum_;            // in pA
+    double threshold_;                 // voltage threshold in mV
+
+    double I_;                 // external current in pA
+    double I_syn_;             // post synaptic current in pA
+    std::vector< double > y1_; // synapse current evolution state 1 in pA
+    std::vector< double > y2_; // synapse current evolution state 2 in pA
+
+    State_( const Parameters_& );
+
+    void get( DictionaryDatum&, const Parameters_& ) const;
+    void set( const DictionaryDatum&, const Parameters_&, double );
+  };
+
+
+  struct Buffers_
+  {
+    Buffers_( glif_psc& );
+    Buffers_( const Buffers_&, glif_psc& );
+
+    std::vector< nest::RingBuffer > spikes_; //!< Buffer incoming spikes through delay, as sum
+    nest::RingBuffer currents_;              //!< Buffer incoming currents through delay,
+
+    //! Logger for all analog data
+    nest::UniversalDataLogger< glif_psc > logger_;
+  };
+
+  struct Variables_
+  {
+    double t_ref_remaining_; // counter during refractory period, in ms
+    double t_ref_total_;     // total time of refractory period, in ms
```

Comment:
`t_ref_ ` is already in `P_`, why repeat it here? It would make sense if it was the refractory time rounded to steps and the expressed in ms again, see comment on t_ref_remaining_.

Reply:
converted to steps. t_ref_remaining_ has been removed.